### PR TITLE
Fix globals, part 6: `performance`

### PR DESCRIPTION
**Pull request by skyclouds2001**
/cc @wbamberg 

part of the #33019

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,5 +60,6 @@
   },
   "devDependencies": {
     "jest": "^29.7.0"
-  }
+  },
+  "packageManager": "yarn@1.22.19+sha1.4ba7fc5c6e704fce2066ecbfb0b0d8976fe62447"
```
Comment (skyclouds2001) [inline package.json:64]:
```suggestion
  }
```

File: files/en-us/web/api/performance_api/performance_data/index.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ For some metrics you do have to tell the browser what to measure:
 
 ## Performance data structure
 
-The Performance API collects performance data for each global object (`window.performance` or `self.performance` in workers). See [`self.performance`](/en-US/docs/Web/API/performance_property) for which APIs are available in window and worker contexts. If you are collecting performance metrics for multiple contexts, also take a look at [`performance.timeOrigin`](/en-US/docs/Web/API/Performance/timeOrigin) to synchronize time origins between contexts.
+The Performance API collects performance data for each global object (`window.performance` or `self.performance` in workers). See {{domxref("window.performance")}} for which APIs are available in window and worker contexts. If you are collecting performance metrics for multiple contexts, also take a look at {{domxref("performance.timeOrigin")}} to synchronize time origins between contexts.
```
Comment (wbamberg) [inline files/en-us/web/api/performance_api/performance_data/index.md:25]:
```suggestion
With the Performance API you can collect performance data in both {{domxref("Window.performance", "Window")}} and {{domxref("WorkerGlobalScope.performance", "Worker")}} global contexts. If you are collecting performance metrics for multiple contexts, take a look at {{domxref("performance.timeOrigin")}} to synchronize time origins between contexts.
```

File: files/en-us/web/api/window/performance/index.md [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+slug: Web/API/Window/performance
+page-type: web-api-instance-property
 browser-compat: api.performance
 ---
 
-{{APIRef("Performance API")}}{{AvailableInWorkers}}
+{{APIRef("Performance API")}}
 
-The global **`performance`** property returns a {{domxref("Performance")}} object, which can be used to gather performance information about the context it is called in (window or worker).
+The **`performance`** property of the {{domxref("Window")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about the context it is called in window.
```
Comment (wbamberg) [inline files/en-us/web/api/window/performance/index.md:11]:
```suggestion
The **`performance`** property of the {{domxref("Window")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about code running in the window's scope.
```

File: files/en-us/web/api/window/performance/index.md [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 
 Performance entries are per context. If you create a mark on the main thread (or other worker), you cannot see it in a worker thread, and vice versa.
 
 ## Value
 
-A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on (window or worker).
+A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on window.
```
Comment (wbamberg) [inline files/en-us/web/api/window/performance/index.md:17]:
```suggestion
A {{domxref("Performance")}} object offering access to performance and timing-related information about code running in the window's scope.
```

File: files/en-us/web/api/workerglobalscope/performance/index.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+title: "WorkerGlobalScope: performance property"
+short-title: performance
+slug: Web/API/WorkerGlobalScope/performance
+page-type: web-api-instance-property
+browser-compat: api.performance
+---
+
+{{APIRef("Performance API")}}
+
+The **`performance`** property of the {{domxref("WorkerGlobalScope")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about the context it is called in worker.
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/performance/index.md:11]:
```suggestion
The **`performance`** property of the {{domxref("WorkerGlobalScope")}} interface returns a {{domxref("Performance")}} object, which can be used to gather performance information about code running in the worker's scope.
```

File: files/en-us/web/api/workerglobalscope/performance/index.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Performance entries are per context. If you create a mark on the main thread (or other worker), you cannot see it in a worker thread, and vice versa.
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/performance/index.md:13]:
```suggestion
Performance entries are per context. If you create a mark on a worker thread, you will not see it in the main thread or any other workers.
```

File: files/en-us/web/api/workerglobalscope/performance/index.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Value
+
+A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on worker.
+
+## Performance API availability
+
+See the table which provides an overview about the availability of the performance APIs in window and worker contexts at {{domxref("Window.performance")}}.
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/performance/index.md:21]: We should not do this. I think we should scrap that table in the Window page, and instead list in this page the APIs that are available to workers, something like:

```
Note that only the following performance API are available in worker contexts:
...list available APIs
```

....and we should put it in the preamble, just before the "Value" H2.

File: files/en-us/web/api/window/performance/index.md [hunk 8]
```diff
@@ -1,20 +1,20 @@
...
 
 ## Performance API availability
```
Comment (wbamberg) [inline files/en-us/web/api/window/performance/index.md:19]: We shouldn't this section here - it's useless for people reading about `Window.performance`. Instead, in `Worker.performance`, we should list the APIs that _are_ supported in workers.

File: files/en-us/web/api/performance/index.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ browser-compat: api.Performance
 
 The **`Performance`** interface provides access to performance-related information for the current page.
 
-An object of this type can be obtained by calling `window.performance` or `self.performance` in workers. Note that Performance entries are per context. If you create a mark on the main thread (or other worker), you cannot see it in a worker thread, and vice versa. See [`self.performance`](/en-US/docs/Web/API/performance_property) for which APIs are available in window and worker contexts.
+An object of this type can be obtained by calling `window.performance` or `self.performance` in workers. Note that Performance entries are per context. If you create a mark on the main thread (or other worker), you cannot see it in a worker thread, and vice versa. See {{domxref("window.performance")}} for which APIs are available in window and worker contexts.
```
Comment (wbamberg) [inline files/en-us/web/api/performance/index.md:12]:
```suggestion
Performance entries are specific to each execution context. You can access performance information for code running in a window via {{domxref("Window.performance")}}, and for code running in a worker via {{domxref("WorkerGlobalScope.performance")}}.
```

**Review by wbamberg — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/workerglobalscope/index.md [hunk 10]
```diff
@@ -37,7 +37,7 @@ _This interface inherits properties from the {{domxref("EventTarget")}} interfac
   - : Returns the {{domxref("WorkerNavigator")}} associated with the worker. It is a specific navigator object, mostly a subset of the {{domxref("Navigator")}} for browsing scopes, but adapted to workers.
 - {{domxref("WorkerGlobalScope.origin")}} {{ReadOnlyInline}}
   - : Returns the global object's origin, serialized as a string.
-- {{domxref("performance_property", "WorkerGlobalScope.performance")}} {{ReadOnlyInline}}
+- {{domxref("WorkerGlobalScope.performance")}} {{ReadOnlyInline}}
   - : Returns the {{domxref("Performance")}} associated with the worker. It is a regular performance object, except that only a subset of its property and methods are available to workers.
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/index.md:41]:
```suggestion
  - : Returns the {{domxref("Performance")}} associated with the worker. Only a subset of the  properties and methods of the `Performance` interface are available to workers.
```

File: files/en-us/web/api/workerglobalscope/performance/index.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+- {{domxref("PerformanceMark")}}
+- {{domxref("PerformanceMeasure")}}
+- {{domxref("PerformanceObserver")}}
+- {{domxref("PerformanceObserverEntryList")}}
+- {{domxref("PerformanceResourceTiming")}}
+- {{domxref("PerformanceServerTiming")}}
+
+## Value
+
+A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on worker.
```
Comment (wbamberg) [inline files/en-us/web/api/workerglobalscope/performance/index.md:28]:
```suggestion
A {{domxref("Performance")}} object offering access to performance and timing-related information for the context it is called on.
```

**Review by wbamberg — APPROVED** [review]
👍 Looks great, thank you @skyclouds2001 !

**Pull request merged**